### DISCUSSION TITLE
Improvement for Matrix Fields: Use of Field Editor Instead of Clip View

### DISCUSSION THREAD

**Discussion by alexanderbuergin** [craftcms/cms#15895] [General]
### Description

I propose changing the input interface for Matrix fields from the current Clip view to the Field Editor, which is already used for all entry types.

### Background

Matrix fields have essentially become similar to entry types, as they can now contain their own fields. By leveraging the existing Field Editor, which is already integrated into the system, the user experience (UX) can be significantly improved.

### Advantages

- **Improved Clarity**: Matrix fields with many different input types (e.g., 15–20 or more) quickly become cluttered. By using tabs and the search functionality available in the Field Editor, this issue could be easily resolved.
- **Reusability of Existing Features**: Instead of developing a new UI component, we could rely on the existing Field Editor and its proven features, saving development time and ensuring consistency within the UI.
- **Enhanced Structure**: The Field Editor allows for logical grouping of fields, which would be especially helpful in managing large Matrix fields with many input masks (e.g., 25+ entry types).

### Current Issues

- Matrix fields with many input types quickly become overwhelming, especially when they should be divided into groups. This issue remains unresolved.
- A plugin that partially addresses this issue already exists for Craft 5, but I believe this functionality should be part of the core of Craft CMS 5, particularly since Craft 5 is focusing on UX improvements in the backend.

This approach could solve many of the existing problems by utilizing already available elements and functionality.

### Expected Behavior

By integrating the Field Editor for Matrix fields, the management and input of fields will become more intuitive, organized, and efficient. Grouping and searching will be directly integrated.

### Images

<img width="1150" alt="Bildschirmfoto 2024-10-16 um 11 32 33" src="https://github.com/user-attachments/assets/e44ac9cc-01c7-4282-9751-5b4c284775c3">

<img width="1178" alt="Bildschirmfoto 2024-10-16 um 11 47 32" src="https://github.com/user-attachments/assets/54463ca5-db8c-4c51-8edb-5b0ec1e55e08">

<img width="449" alt="Bildschirmfoto 2024-10-16 um 11 49 08" src="https://github.com/user-attachments/assets/0d6fa22b-b229-42bb-bf8a-f2b1a634dc1e">

**Comment by davidhellmann** [comment]
<img width="685" alt="image" src="https://github.com/user-attachments/assets/e8414886-daa3-408a-8b13-b26a08b7dcb6">
This should also be in the right color and have icons. Otherwise, agree!

**Comment by thupsi** [comment]
Nice idea!

**Comment by SkermBE** [comment]
I also feel the need for this. Some projects have lots of different fields, feels clumsy now.
A simple tab to organise the fields like MatrixMate or Neo had/has feels like a needed addition.

<img width="202" alt="Screenshot 2025-01-29 at 22 31 37" src="https://github.com/user-attachments/assets/2f55bdbb-ce10-48fd-914f-2114c91bdcbf" />

**Reply by alexanderbuergin** [reply]
<3

**Comment by brandonkelly** [comment]
Not the same UI, but Craft 5.8.0 is out with the ability to organize Matrix fields’ entry types into groups. (#17425)

**Comment by brandonkelly** [comment]
The “Add entry” button has been expanded to a button group for Matrix fields with multiple groups, for Craft 5.9 (#17731).

**Reply by alexanderloewe** [reply]
Love that! Thanks

**Reply by brandonkelly** [reply]
Craft 5.9.0 is out now with that change 🎉
